latch/services/execute.py

```diff
@@ -0,0 +1,204 @@
+"""Service to execute a workflow."""
+
+import json
+import typing
+from pathlib import Path
+from types import ModuleType
+from typing import Union
+
+import google.protobuf.json_format as gpjson
+import requests
+from flyteidl.core.types_pb2 import LiteralType
+from flytekit.core.context_manager import FlyteContextManager
+from flytekit.core.type_engine import TypeEngine
+from latch.utils import retrieve_or_login
+
+
+def execute(params_file: Path, version: Union[None, str] = None) -> bool:
+    """Executes a versioned workflow with parameters specified in python.
+
+    Args:
+        params_file: A path pointing to a python file with a function call
+            representing the workflow execution with valid parameter values.
+        version: An optional workflow version to execute, defaults to latest if
+            None is provided
+
+        Parameters for the workflow execution are specified in plain python as
+        a function call, as the syntax is highly legible and easier to work
+        with than alternatives such as YAML, JSON. This file is passed to the
+        `execute` service and should contain no code other than the single
+        call.
+
+    Returns:
+        True if successful else False
```

Comment:
```suggestion
        True if successful
```

this phrasing is suggested somewhere in google's style guide